plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/models.py

```diff
@@ -357,12 +357,19 @@ async def soft_delete(self, session: ProfileSession):
         Soft delete the tenant record by setting its state to 'deleted'.
         Note: This method should be called on an instance of the TenantRecord.
         """
+        # Delete api records
+        recs = await TenantAuthenticationApiRecord.query_by_tenant_id(
+            session, self.tenant_id
+        )
+        for rec in recs:
```

Comment:
I don't mind this explicit for loop but just fyi you could use list comprehension here if you wanted.

Reply:
Is there a reason to use list comprehension here if we don't intend on using the list for anything? I may be misunderstanding

Reply:
Just if you want to write it in one line. You don't have to use the returned list. Sometimes it's more readable. But I don't know, people have different opinions.